tests/test_imports.py

```diff
@@ -0,0 +1,41 @@
+import subprocess
+import sys
+
+import pytest
+
+
+@pytest.mark.skipif("starlite" in sys.modules, reason="This tests expects starlite to be imported for the first time")
+def test_testing_import_deprecation() -> None:
+    import starlite
+
+    dynamic_imports_names = {"TestClient", "create_test_client", "create_test_request"}
+    assert dynamic_imports_names & starlite.__dict__.keys() == set()
+    assert "starlite.testing" not in sys.modules
+    assert "requests" not in sys.modules
+    assert "requests.models" not in sys.modules
+
+    dynamic_imports = {}
+    for dynamic_import_name in dynamic_imports_names:
+        dynamic_imports[dynamic_import_name] = getattr(starlite, dynamic_import_name)
+
+    assert dynamic_imports_names & starlite.__dict__.keys() == dynamic_imports_names
+    assert "starlite.testing" in sys.modules
+    assert "requests.models" in sys.modules
+
+    from starlite import testing
+
+    assert dynamic_imports == {
+        "TestClient": testing.TestClient,
+        "create_test_client": testing.create_test_client,
+        "create_test_request": testing.create_test_request,
+    }
+
+
+@pytest.mark.skipif(
+    "starlite" not in sys.modules, reason="Was able to run previous test, no need to launch subprocess for that"
+)
+def test_testing_import_deprecation_in_subprocess() -> None:
```

Comment:
You can set `starite` to `None` and achieve the same result. Example: https://github.com/vrslev/ikea-api-client/blob/1b9003111ed774751705ed14076dbb0882a24bd0/tests/test_init.py

Reply:
Source code: https://github.com/vrslev/ikea-api-client/blob/1b9003111ed774751705ed14076dbb0882a24bd0/src/ikea_api/__init__.py

Reply:
For some reason, It worked If I ran test separately, but not when they all ran with pytest. I tried `importlib.reload()` as well.

Having a clean subprocess is slower, but more right in terms of what we expect to get. Reloading modules when they already were imported somewhere may cause weird bugs when `some_module.reloaded_module.Class is not reloaded_module.Class` if `some_module` imported `reloaded_module` before it was reloaded and check is happening after.

Reply:
To be more specific:

`reloaded_module.py`
```py
class Class:
    pass
```

`some_module.py`
```py
from reloaded_module import Class

instance = Class()
```

`main.py`
```py
import sys
import reloaded_module
import some_module

assert isinstance(some_module.instance, reloaded_module.Class), "Before reload"

del sys.modules["reloaded_module"]
import reloaded_module

assert isinstance(some_module.instance, reloaded_module.Class), 'After reload'
```

Running main.py will result in
```py
Traceback (most recent call last):
  File "main.py", line 10, in <module>
    assert isinstance(some_module.instance, reloaded_module.Class), 'After reload'
AssertionError: After reload
```

Reply:
I don’t think it matters because tests are isolated from each other 🤔

Reply:
I wouldn't say tests are really isolated. To some degree, maybe. But source code isn't.

Tests are collected first, then executed. Almost all of the imports are done during collection phase.
If then one module is reloaded, there still might be a conflict between what's already imported in other tests modules later in execution.

If it's still unclear what I mean, I'm pretty sure I can come up with another example.

Reply:
No, I get it, thanks :)